src/crypto/crypto_mbedtls_alt.c

```diff
@@ -1658,7 +1662,7 @@ static mbedtls_ecp_group_id crypto_mbedtls_ecp_group_id_from_ike_id(int group)
     }
 }
 
-#ifdef CRYPTO_MBEDTLS_CRYPTO_EC
+#if defined(CRYPTO_MBEDTLS_CRYPTO_EC) && (defined(CONFIG_DPP) || defined(CONFIG_SAE_PK))
```

Comment:
there are too many CONFIG_DPP || CONFIG_SAE_PK in this file. can you put them together if they are for same usage?
And CRYPTO_MBEDTLS_CRYPTO_EC is currently defined by default. can you use
#ifdef CONFIG_DPP || CONFIG_SAE_PK
#define CRYPTO_MBEDTLS_CRYPTO_EC
#endif
to make it cleaner to read?

Reply:
I will check this. currently `CRYPTO_MBEDTLS_CRYPTO_EC` is not under any check.

Reply:
Earlier all functions were enabled by default as `CRYPTO_MBEDTLS_CRYPTO_EC` was always enabled. currently, I have placed it under the below check as some functions are needed for `EAP_PWD` , `EAP_SERVER_PWD` or `CONFIG_SAE` .

#if defined(CONFIG_DPP) || defined(CONFIG_SAE_PK) || defined(EAP_PWD) \
    || defined(EAP_SERVER_PWD) || defined(CONFIG_SAE)
#define CRYPTO_MBEDTLS_CRYPTO_EC
#endif

Reply:
can (defined(CONFIG_DPP) || defined(CONFIG_SAE_PK)) be removed then?

Reply:
Yes - it is removed. Pushing the changes now